File: 02-netcore/01-examples/04-get-and-exception-handling/tools/diagnostics/error/ErrorHandler.cs
namespace _04_get_and_exception_handling.Diagnostics
{
    class ErrorHandler
    {
        private System.Exception _errorInformation;
        public readonly int StatusCode;

        public ErrorHandler(int statusCode, System.Exception errorInformation){
            StatusCode = statusCode;
            _errorInformation = errorInformation;
        }
            public string ConsoleResponse()
            {
                var message = string.Format("Status:{0}, Message:{1}, StackTrace:{2}", StatusCode, _errorInformation.Message, _errorInformation.StackTrace); 

                return message;
            }

            public HttpErrorModel HttpResponse()
            {
                var httpError = new HttpErrorModel {StatusCode = StatusCode, Message = "Internal error", Type = "Internal"};
               
                return httpError;
            }
        }
}

File: 03-python/01-examples/04-get-and-exception-handling/tools/diagnostics/error/errorhandler.py
class ErrorHandler():
    def __init__(self, error):
        super().__init__()
        self.message = error.message
        self.status = error.status
        self.stacktrace = error.stacktrace

    def sendToConsole(self):
        print(" * Message: {0}".format(self.message))
        print(" * StatusCode: {0}".format(self.status))
        print(" * StackTrace: {0}".format(self.stacktrace))

    def httpResponse(self):
        response = dict()
        response['Message'] = self.message
        response['StatusCode'] = self.status
        response['Type'] = 'internal'

        return response